gpytorch/lazy/lazy_tensor.py

```diff
@@ -1585,15 +1621,33 @@ def zero_mean_mvn_samples(self, num_samples):
             :obj:`torch.tensor`:
                 Samples from MVN (num_samples x batch_size x num_dim) or (num_samples x num_dim)
         """
-        if self.size()[-2:] == torch.Size([1, 1]):
-            covar_root = self.evaluate().sqrt()
+        if settings.ciq_samples.on():
+            from ..utils.contour_integral_quad import contour_integral_quad
+
+            base_samples = torch.randn(
+                *self.batch_shape, self.size(-1), num_samples, dtype=self.dtype, device=self.device
+            )
+            base_samples = base_samples.permute(-1, *range(self.dim() - 1)).contiguous()
+            base_samples = base_samples.unsqueeze(-1)
+            solves, weights, _, _ = contour_integral_quad(
+                self.evaluate_kernel(),
```

Comment:
what's the plan for custom mvm support?

Reply:
This just evaluates any outstanding kernel calls, it doesn't necessarily explicitly materialize an `n x n` matrix (i.e., `evaluate_kernel` vs `evaluate`).

Reply:
ah gotcha thanks